src/Solver/Dynamic/Ida.cpp

```diff
@@ -248,7 +223,18 @@ namespace AnalysisManager
       {
         retval = IDASolve(solver_, tout, &tret, yy_, yp_, IDA_NORMAL);
         checkOutput(retval, "IDASolve");
-        // printOutput(tout);
+
+        if (step_callback.has_value())
+        {
+          // The callback may try to observe upated values in the model, so we should update them here
+          // (At this point, the model's values are one internal integrator step out of date)
+          model_->updateTime(tret, 0.0);
+          copyVec(yy_, model_->y());
```

Comment:
I believe the model time here should be set to `tout`
```suggestion
          model_->updateTime(tout, 0.0);
          copyVec(yy_, model_->y());
```

Reply:
If everything goes well with IDA then `tout == tret` however `tret` is the time that IDA is asserting the current state is at - if anything happens to go wrong I'd rather trust what time IDA says it is than what time I'm assuming it should be. According to [the IDA documentation](https://sundials.readthedocs.io/en/latest/ida/Usage/index.html#c.IDASolve), there are 3 different reasons IDA can return successfully:
- `IDA_SUCCESS`: we reached `tout` and returned - this is what normally happens,
- `IDA_TSTOP_RETURN`: a stop time earlier than `tout` was configured and we stopped early - then if I use `tout` instead of `tret` this is a bug.
- `IDA_ROOT_RETURN`: IDA was configured to search for roots and found one - then if I use `tout` instead of `tret` this is a bug.

I don't think we'll ever use those configuration options but the way I see it using `tret` in this situation is just more future-proof than using `tout`